doc/source/tutorial/stats/sampling.rst

```diff
@@ -0,0 +1,199 @@
+* The Ratio-of-Uniforms Method: This is a type of acceptance-rejection
+  method which is uses minimal bounding rectangles to construct the hat
+  function. See :func:`~rvs_ratio_uniforms`.
+* Inversion for Discrete Distributions: The difference compared to the
+  continuous case is that :math:`F` is now a step-function. To realize
+  this in a computer, a search algorithm is used, the simplest of which
+  is *sequential search*. A uniform random number is generated from
+  U(0, 1) and probabilities are summed until the cumulative probability
+  exceeds the uniform random number. The index at which this happens is
+  the required random variate and is returned.
+
+
+More details on these algorithms can be found in the `appendix of the UNU.RAN
+user manual <http://statmath.wu.ac.at/software/unuran/doc/unuran.html#RVG>`__.
+
+
+Basic concepts of the Interface
+-------------------------------
+
+Every generator needs to be set up before one can start sampling from it.
+This can be done by instantiating an object of that class. Most of the
+generators take a distribution object as input which contains the implementation
+of required methods like PDF, CDF, etc. In addition to the distribution
+object, one can also pass parameters used to set up the generator. It is also
+possible to truncate the distributions using a ``domain`` parameter.  All
+generators need a stream of uniform random numbers that are transformed into
+random variates of the given distribution. This is done by passing a ``seed``
+parameter with a NumPy BitGenerator as the uniform random number generator.
+``seed`` can either be a integer, `np.random.Generator`,
+`np.random.BitGenerator`, `np.random.RandomState`, or
+`np.random.SeedSequence`.
+
+.. warning:: Use of NumPy < 1.19.0 is discouraged as it doesn't have a fast
+             Cython API for generating uniform random numbers and might be
+             too slow to practically rely on.
+
+All the generators have a common ``rvs`` method that can be used to draw
+samples from the given distribution.
+
+An example of this interface is shown below:
+
+    >>> from scipy.stats import TransformedDensityRejection
+    >>> from math import exp
+    >>> 
+    >>> class StandardNormal:
+    ...     def pdf(self, x: float) -> float:
+    ...         # note that the normalization constant isn't required
+    ...         return exp(-0.5 * x*x)
+    ...     def dpdf(self, x: float) -> float:
+    ...         return -x * exp(-0.5 * x*x)
+    ... 
+    >>> dist = StandardNormal()
+    >>> 
+    >>> urng = np.random.default_rng(0x6be00a9336fc82258fe73bfa672d3d52)
+    >>> rng = TransformedDensityRejection(dist, seed=urng)
+
+As shown in the example, we first initialize a distribution object that
+contains an implementation of the methods required by the generator. In
+our case, we use the :class:`~TransformedDensityRejection` method which
+requires a PDF and its derivative w.r.t. x (i.e. the variate). Note that
```

Comment:
should we highlight that point even more (e.g., in bold or .. note :: ...) and point out that it applies to the other methods like dpdf, too?

Reply:
> should we highlight that point even more (e.g., in bold or .. note :: ...) and point out that it applies to the other methods like dpdf, too?

This is a very good point. I have made this a note now.